src/ConnectedMode/Migration/IMigrationPrompt.cs

```diff
@@ -34,6 +32,10 @@ namespace SonarLint.VisualStudio.ConnectedMode.Migration
     /// <summary>
     /// In charge of showing users a prompt to migrate to a new connected mode.
     /// </summary>
+    /// <remarks>The caller can assume that the component follows VS threading rules
+    /// i.e. the implementing class is responsible for switching to the UI thread if necessary.
+    /// The caller doesn't need to worry about it.
+    /// </remarks>
```

Comment:
I've added this comment to all three of the interfaces I've touched as part of this PR, since the implementations all handle VS threading correctly.